grpc-testtool/src/bin/process-server.rs

```diff
@@ -3,82 +3,72 @@
-use env_logger::{Builder, Target};
+use env_logger::Builder;
 use log::{info, LevelFilter};
 use rpc::{
     process_server::process_server_service_server::ProcessServerServiceServer,
     rpcdb::database_server::DatabaseServer as RpcServer, sync::db_server::DbServer as SyncServer,
     DatabaseService,
 };
+use serde::Deserialize;
 use std::{
     error::Error,
     io::Write,
     net::{IpAddr::V4, Ipv4Addr},
     path::PathBuf,
+    str::FromStr,
     sync::Arc,
 };
-use tempdir::TempDir;
 use tonic::transport::Server;
 
+#[derive(Clone, Debug, Deserialize)]
+struct Options {
+    #[serde(default = "Options::history_length_default")]
+    history_length: u32,
+}
+
+impl Options {
+    // used in two cases:
+    //  serde deserializes Options and there was no history_length
+    // OR
+    //  Options was not present
+    const fn history_length_default() -> u32 {
+        100
+    }
+}
+
+impl FromStr for Options {
+    type Err = String;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        serde_json::from_str(s).map_err(|e| format!("error parsing options: {}", e))
+    }
+}
+
 /// A GRPC server that can be plugged into the generic testing framework for merkledb
 
 #[derive(Debug, Parser)]
 #[command(author, version, about, long_about = None)]
 struct Opts {
-    #[arg(short = 'g', long, default_value_t = 10000)]
+    #[arg(short, long)]
     //// Port gRPC server listens on
     grpc_port: u16,
 
-    #[arg(short = 'G', long, default_value_t = 10001)]
-    /// Port gRPC gateway server, which HTTP requests can be made to
-    _gateway_port: u16,
-
-    #[arg(short = 'l', long, default_value = temp_path())]
-    log_dir: PathBuf,
-
-    #[arg(short = 'L', long, default_value_t = LevelFilter::Info)]
-    log_level: LevelFilter,
-
-    #[arg(short = 'b', long)]
-    _branch_factor: Option<u16>,
-
-    #[arg(short = 'p', long)]
-    _process_name: String,
-
-    #[arg(short = 'd')]
+    #[arg(short, long)]
     db_dir: PathBuf,
 
-    #[arg(short = 'H')]
-    _history_length: Option<u32>,
-
-    #[arg(short = 'N')]
-    _node_cache_size: Option<u32>,
-}
+    #[arg(short, long, default_value_t = LevelFilter::Info)]
+    log_level: LevelFilter,
 
-fn temp_path() -> clap::builder::OsStr {
-    let tmpdir = TempDir::new("process-server").expect("unable to create temporary directory");
-    // we leak because we want the temporary directory to stick around forever (emulating what happens in golang)
-    Box::leak(Box::new(tmpdir.into_path())).as_os_str().into()
+    #[arg(short, long)]
+    config: Option<Options>,
 }
 
 #[tokio::main]
 async fn main() -> Result<(), Box<dyn Error>> {
     // parse command line options
     let args = Opts::parse();
 
-    match std::fs::create_dir_all(&args.log_dir) {
-        Ok(()) => {}
-        Err(e) if e.kind() == std::io::ErrorKind::AlreadyExists => {}
-        Err(e) => panic!("Unable to create directory\n{}", e),
-    }
-
-    // set up the log file
-    let logfile = std::fs::OpenOptions::new()
```

Comment:
Why do we remove the part to log to file?

Reply:
Because the orchestrator now captures stdout and stderr